src/main/java/io/nats/client/Options.java

```diff
@@ -2500,10 +2499,11 @@ private static void appendOption(CharBuffer builder, String key, String value, b
         _appendOptionEnd(builder, quotes);
     }
 
-    private static void appendOption(CharBuffer builder, String key, char[] value, boolean quotes, boolean comma) {
-        _appendStart(builder, key, quotes, comma);
+    @SuppressWarnings("SameParameterValue")
+    private static void appendOption(CharBuffer builder, String key, char[] value, boolean comma) {

```

Comment:
When used for a char array, it always represents a string value, so must be in quotes